src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
I'm confident that the complete refactoring will still be manageable in size, so I'd probably go for it all-at-once in the same PR. 

Having said this, I propose to merge this PR and open a new one for the tech-debt, to ensure we're shipping the feature in time for 0.16. @aldipower @orlandohohmeier OK?